### ISSUE TITLE
Multi-phase workflows with automated commits and PR

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#305]
### TL;DR

In 2024 we intend to automate more parts of the user journey around using the abc CLI. Currently the CLI just spits out files onto the local filesystem, and the user is responsible for:

 - run linting and autoformatting
 - run terraform init
 - git commit
 - git push
 - create github PR
 - merge github PR
 - chaining multiple templates together (e.g. first run the template to create the GCP folders, then run the template to create the projects, then ...etc)

We can do more to automate this. The abc CLI (or some bot running in github) can handle much or all of it.

We may end up dropping Windows support during all this rather than absorb the extra complexity of running commands in two different ways.

cc @verbanicm @dcreey @kculligan @sethvargo @mattrandallbecker
